### PR TITLE
fix(argo-cd): Added openshift route options to values.yaml

### PR DESCRIPTION
**Pull request by bbensky**
I would like to add Openshift route properties to the values.yaml file that reflect existing options in the [route.yaml ](https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/templates/argocd-server/route.yaml) template.

Checklist:

* [X] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [X] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [X] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Review by mbevc1 — COMMENTED** [review]
Looks good, we probably want to remove defaults in the template https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/templates/argocd-server/route.yaml#L21-L22 as we're setting this here :thinking:

**Comment by bbensky** [comment]
> 
> 
> Looks good, we probably want to remove defaults in the template https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/templates/argocd-server/route.yaml#L21-L22 as we're setting this here 🤔

Thanks - I removed them from the template.

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -999,10 +999,16 @@ server:
...
-    # -- Enable a OpenShift route for the Argo CD server
+    # -- Enable an OpenShift Route for the Argo CD server
     enabled: false
-    # -- Hostname of OpenShift route
+    # -- Openshift Route annotations
+    annotations: {}
+    # -- Hostname of OpenShift Route
     hostname: ""
+    # -- Termination type of Openshift Route
+    termination_type: Passthrough
```
Comment (mkilchhofer) [inline charts/argo-cd/values.yaml:1009]: Are you sure that uppercase fits here? Before your change it was lower case and even in the OpenShift [here](https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html) docs it is lower case:
```yaml
apiVersion: v1
kind: Route
metadata:
  name: route-edge-secured 
spec:
  host: www.example.com
  to:
    kind: Service
    name: service-name 
  tls:
    termination: edge            
    key: |-                      
      -----BEGIN PRIVATE KEY-----
      [...]
      -----END PRIVATE KEY-----
    certificate: |-              
      -----BEGIN CERTIFICATE-----
      [...]
      -----END CERTIFICATE-----
    caCertificate: |-            
      -----BEGIN CERTIFICATE-----
      [...]
      -----END CERTIFICATE-----
---
apiVersion: v1
kind: Route
metadata:
  name: route-passthrough-secured 
spec:
  host: www.example.com
  to:
    kind: Service
    name: service-name 
  tls:
    termination: passthrough
---
apiVersion: v1
kind: Route
metadata:
  name: route-pt-secured 
spec:
  host: www.example.com
  to:
    kind: Service
    name: service-name 
  tls:
    termination: reencrypt        
    key: [as in edge termination]
    certificate: [as in edge termination]
    caCertificate: [as in edge termination]
    destinationCACertificate: |-  
      -----BEGIN CERTIFICATE-----
      [...]
      -----END CERTIFICATE-----
```

**Review by mkilchhofer — APPROVED** [review]

**Review by mbevc1 — APPROVED** [review]

**Comment by mbevc1** [comment]
Thanks for your contribution, looks good now :+1:

**Pull request merged**